### ISSUE TITLE
[Docs/Core] Cannot run processes in background with nohup

### ISSUE THREAD

**Issue by romilbhardwaj** [skypilot-org/skypilot#3803]
A common pattern to run processes in the background (e.g., run mounting daemon, logging etc.) is to use `nohup mycmd &`.

Use of `nohup` in SkyPilot `run` or `setup` leaves it stuck waiting for the process to complete instead of running in the background.

Example YAML:
```
setup: |
  # Test if nohup works
  echo "Starting setup"
  nohup sleep 100000 &
  echo "Setup done"

run: |
  echo "Starting run"
  nohup sleep 100000 &
  echo "Run done"
```

`sky launch` on this will wait for the sleep to complete:
```
I 07-31 15:51:57 provisioner.py:550] Successfully provisioned cluster: test
I 07-31 15:51:57 cloud_vm_ray_backend.py:3162] Running setup on 1 node.
Starting setup
Setup done
<stays stuck here>
```

**Comment by alex000kim** [comment]
Yup, I am running into this issue with a vLLM benchmarking script https://github.com/vllm-project/vllm/blob/main/benchmarks/benchmark_serving.py
that requires starting a vllm server first (with nohup) and then running that script.

**Comment by romilbhardwaj** [comment]
BTW, this happens because `nohup` stays connected to the stdout or stderr and our underlying SSH will keep the connection open to handle any potential output from the command.

To workaround, redirect your command's output with `> /dev/null 2>&1`:

```
nohup mycmd > /dev/null 2>&1 &
```

You can change /dev/null to a file to keep your output for later viewing.

**Comment by alex000kim** [comment]
Oh gotcha! Thanks for the workaround!

**Comment by Michaelvll** [comment]
We should add this to FAQ section of the docs.
